lib/ood_core/job/adapters/slurm.rb

```diff
@@ -89,22 +95,66 @@ def initialize(cluster: nil, bin: nil, conf: nil, bin_overrides: {})
           #   #  ...
           #   #]
           # @param id [#to_s] the id of the job
-          # @param filters [Array<Symbol>] list of attributes to filter on
+          # @param attrs [Array<Symbol>, nil] list of attributes request when calling squeue
           # @raise [Error] if `squeue` command exited unsuccessfully
           # @return [Array<Hash>] list of details for jobs
-          def get_jobs(id: "", filters: [])
-            delim = "\x1F"     # don't use "|" because FEATURES uses this
-            options = filters.empty? ? fields : fields.slice(*filters)
-            args  = ["--all", "--states=all", "--noconvert"]
-            args += ["-o", "#{options.values.join(delim)}"]
-            args += ["-j", id.to_s] unless id.to_s.empty?
-            lines = call("squeue", *args).split("\n").map(&:strip)
+          def get_jobs(id: "", attrs: nil, record_separator: RECORD_SEPARATOR, unit_separator: UNIT_SEPARATOR)
+            fields = squeue_fields(attrs)
+            args = squeue_args(id: id, options: fields.values, record_separator: record_separator, unit_separator: unit_separator)
+
+            #TODO: switch mock of Open3 to be the squeue mock script
+            # then you can use that for performance metrics
+            StringIO.open(call("squeue", *args)) do |output|
+              # if cluster, skip 2 lines, else skip 1 line
+              (cluster ? 2 : 1).times { output.gets(RECORD_SEPARATOR) }
```

Comment:
I think you meant this to be the method parameter `record_separator` instead of the constant `RECORD_SEPARATOR`.

Also, why parameterize for that? Do you anticipate that we may need to vary the separator characters? If so then why use constants?

Reply:
Premature optimization. Thanks. Removed!